### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by bohdan-mykhailenko**
[DEMO LINK](https://bohdan-mykhailenko.github.io/react_movies-list-add-form/)

### CODE REVIEW

**Review by silchyn — CHANGES_REQUESTED** [review]
- I can submit a form with a title consisting of spaces
- Can we show the URL validation error under the field like the required validation error?

File: src/components/NewMovie/NewMovie.tsx [hunk 1]
```diff
@@ -1,45 +1,108 @@
...
+  imdbUrl: '',
+  imdbId: '',
+};
+
+export const NewMovie: React.FC<NewMovieProps> = ({ addMovie }) => {
+  const [count, setCount] = useState(0);
+  const [formState, setFormState] = useState(initialFormState);
+  const [isInvalidUrl, setIsInvalidUrl] = useState(false);
+  // eslint-disable-next-line max-len
+  const pattern = /^((([A-Za-z]{3,9}:(?:\/\/)?)(?:[-;:&=+$,\w]+@)?[A-Za-z0-9.-]+|(?:www\.|[-;:&=+$,\w]+@)[A-Za-z0-9.-]+)((?:\/[+~%/.\w-_]*)?\??(?:[-+=&;%@,.\w_]*)#?(?:[,.!/\\\w]*))?)$/;
```
Comment (dffUqp) [inline src/components/NewMovie/NewMovie.tsx:22]: let's create a helper function for checking our value.

something like: 
```jsx
const validateUrl = (value) => {
  const pattern = /^((([A-Za-z]{3,9}:(?:\/\/)?)(?:[-;:&=+$,\w]+@)?[A-Za-z0-9.-]+|(?:www\.|[-;:&=+$,\w]+@)[A-Za-z0-9.-]+)((?:\/[+~%/.\w-_]*)?\??(?:[-+=&;%@,.\w_]*)#?(?:[,.!/\\\w]*))?)$/;
  
  return pattern.test(value)
}
 ```

File: src/components/NewMovie/NewMovie.tsx [hunk 2]
```diff
@@ -1,45 +1,108 @@
...
+
+  const {
+    title,
+    description,
+    imgUrl,
+    imdbUrl,
+    imdbId,
+  } = formState;
+
+  const isFormFilled = title && imgUrl && imdbUrl && imdbId;
```
Comment (dffUqp) [inline src/components/NewMovie/NewMovie.tsx:32]: I'm still can submit a form with a title consisting of spaces. 

add `.trim()` to prevent this behavior

**Review by dffUqp — CHANGES_REQUESTED** [review]
LGTM:fire: but needs a few fixes:upside_down_face:

File: src/components/NewMovie/NewMovie.tsx (hunk 2 above)
Comment (dffUqp) [reply src/components/NewMovie/NewMovie.tsx:32]: image

File: src/components/NewMovie/NewMovie.tsx [hunk 3]
```diff
@@ -1,45 +1,107 @@
...
+
+  const {
+    title,
+    description,
+    imgUrl,
+    imdbUrl,
+    imdbId,
+  } = formState;
+
+  const isFormFilled = title.trim() && imgUrl && imdbUrl && imdbId.trim();
```
[the remark this answers is not in the document]
Comment (bohdan-mykhailenko) [reply src/components/NewMovie/NewMovie.tsx:31]: I suit checking of url values in my validateUrl function

File: src/components/NewMovie/NewMovie.tsx [hunk 4]
```diff
@@ -1,45 +1,107 @@
...
+    addMovie(formState);
+
+    setCount((currentCount) => currentCount + 1);
+    setFormState(initialFormState);
+    setIsInvalidUrl(false);
+  };
+
+  const changeFormState = (key: string, value: string) => {
+    const normalizedValue = value
+      .toLowerCase()
```
[the remark this answers is not in the document]
Comment (bohdan-mykhailenko) [reply src/components/NewMovie/NewMovie.tsx:51]: I remove extra spaces to avoid a mess in the text fields

File: src/App.tsx [hunk 5]
```diff
@@ -1,16 +1,25 @@
...
 import { MoviesList } from './components/MoviesList';
 import { NewMovie } from './components/NewMovie';
 import moviesFromServer from './api/movies.json';
+import { Movie } from './types/Movie';
 
 export const App = () => {
+  const [movies, setMovies] = useState(moviesFromServer);
+
+  const addMovie = (movie: Movie) => {
+    setMovies([...movies, movie]);
```
Comment (dffUqp) [inline src/App.tsx:12]: good practice to use a callback when we need a prevState
```suggestion
    setMovies((prevMovies) => [...prevMovies, movie]);
```

**Review by dffUqp — APPROVED** [review]
:fire:

File: src/components/NewMovie/NewMovie.tsx [hunk 6]
```diff
@@ -48,9 +109,16 @@ export const NewMovie = () => {
             type="submit"
             data-cy="submit-button"
             className="button is-link"
+            disabled={!isFormFilled}
           >
             Add
           </button>
+
+          {isInvalidUrl && (
+            <h5 style={{ color: 'red' }}>
```
Comment (dffUqp) [inline src/components/NewMovie/NewMovie.tsx:118]: try to avoid using `inline-styles`

**Pull request closed without merging**